src/Type/Constant/ConstantBooleanType.php

```diff
@@ -30,6 +33,38 @@ public function describe(VerbosityLevel $level): string
 		return $this->value ? 'true' : 'false';
 	}
 
+	public function getSmallerType(bool $orEqual = false): Type
```

Comment:
1) Why is the parameter optional? I'd rather make it required.
2) Or I'd even make it separate methods without any arguments. `getSmallerType(true)` is worse than `getSmallerTypeOrEqual()`.

Reply:
Ah yes, the parameter should be mandatory.

I did consider making it a separate method, but almost always the implementation of `getSmallerType()` will be very similar to `getSmallerOrEqualType()`, so it makes sense to have only one. Alternatively there can be two methods, plus a third private method for the actual implementation, but that might be over-complicating things.

Reply:
I'd say that the simplicity of implementation should not dictate how an interface looks like :) So we should start from having both getSmallerType() and getSmallerOrEqualType(). I'd say that in simple cases we can simply duplicate the common parts. In more complicated cases there can be a third private method, or getSmallerOrEqualType() can simply call getSmallerType() internally and maybe add something to the result for example. It can be different based on the situation.

Once you do this modification, I'll merge this. Thank you!

Reply:
For consistency, perhaps `Type::isSmallerThan()` and `CompoundType::isGreaterThan()` should be split into a separate `OrEqual` variants as well. But I can do a separate PR for that, since those are unrelated to this PR.